pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -933,7 +934,11 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
             currentBalanceTokenOut.sub(amountOut)
         );
 
-        _payProtocolAndManagementFees(normalizedWeights, preSwapBalances, postSwapBalances);
+        uint256 invariantGrowthRatio = FixedPoint.divDown(
+            WeightedMath._calculateInvariant(normalizedWeights, postSwapBalances),
+            WeightedMath._calculateInvariant(normalizedWeights, preSwapBalances)
+        );
```

Comment:
You could simply compute the weighted product of the balance ratios here, saving two calls to `pow`.